### PR TITLE
[5G: MVC][component: agw][type: enhancement] AMF state machine feature support

### PR DESCRIPTION
**Pull request by srinivas-47**
## Summary
AMF state machine Implementation. Handled Following states conversions

UE states:
DEREGISTERED  --> COMMON_PROCEDURE_INITIATED_1
COMMON_PROCEDURE_INITIATED_1 --> COMMON_PROCEDURE_INITIATED_2
COMMON_PROCEDURE_INITIATED_2 --> REGISTERED_CONNECTED
REGISTERED_CONNECTED  --> DEREGISTERED_INITIATED
DEREGISTERED_INITIATED --> DEREGISTERED

PDU session states:
SESSION_NULL --> CREATING
CREATING --> CREATED
CREATED -->RELEASE

## Additional Information

- [ ] This change is backwards-breaking

### CODE REVIEW

File: lte/gateway/c/oai/tasks/amf/Registration.cpp [hunk 1]
```diff
@@ -206,7 +202,16 @@ int amf_proc_registration_request(
     OAILOG_FUNC_RETURN(LOG_AMF_APP, rc);
   }
 
-  rc = amf_registration_run_procedure(&ue_m5gmm_context->amf_context);
+  /*
+   * Execute Initial identity Request from AMF to UE
+   */
+  // rc = state_handle_message_ue_1(
+  //    ue_m5gmm_global_context.mm_state, STATE_EVENT_REG_REQUEST, SESSION_NULL,
+  //    &ue_m5gmm_global_context, &ue_ctx.amf_context);
```
Comment (ulaskozat) [inline lte/gateway/c/oai/tasks/amf/Registration.cpp]: remove unused code

File: lte/gateway/c/oai/tasks/amf/Registration.cpp [hunk 2]
```diff
@@ -206,7 +202,16 @@ int amf_proc_registration_request(
...
+  rc = state_handle_message_ue_1(
```
Comment (ulaskozat) [inline lte/gateway/c/oai/tasks/amf/Registration.cpp]: what is the significance of the suffix '1'?

File: lte/gateway/c/oai/tasks/amf/amf_app_ue_context_and_proc.h [hunk 3]
```diff
@@ -755,4 +748,23 @@ void amf_app_handle_cm_idle_on_ue_context_release(
...
+int state_handle_message_ue_1(
+    m5gmm_state_t cur_state, int event, SMSessionFSMState session_state,
+    ue_m5gmm_context_s* ue_m5gmm_context, amf_context_t* amf_context);
+int state_handle_message_ue_2(
+    m5gmm_state_t, int, SMSessionFSMState, ue_m5gmm_context_s*,
+    amf_ue_ngap_id_t, bstring, int, amf_nas_message_decode_status_t);
+int state_handle_message_ue_3(
+    m5gmm_state_t, int event, SMSessionFSMState, ue_m5gmm_context_s*,
+    amf_ue_ngap_id_t);
+int state_session_handle_message_1(
```
Comment (ulaskozat) [inline lte/gateway/c/oai/tasks/amf/amf_app_ue_context_and_proc.h]: we need to find better names for these before porting to master branch.

**Comment by ulaskozat** [comment]
Please fill in the summary section with details on what is implemented on this PR. Do not repeat what the title already states.

File: lte/gateway/c/oai/tasks/amf/Registration.cpp (hunk 1 above)
Comment (srinivas-47) [reply lte/gateway/c/oai/tasks/amf/Registration.cpp]: Modification done.

File: lte/gateway/c/oai/tasks/amf/Registration.cpp (hunk 2 above)
Comment (srinivas-47) [reply lte/gateway/c/oai/tasks/amf/Registration.cpp]: we are using separate message handle functions for handlers that's why we kept as suffix 1,Now the API names got modified and removed the suffix 1 stuff.

File: lte/gateway/c/oai/tasks/amf/amf_app_ue_context_and_proc.h (hunk 3 above)
Comment (srinivas-47) [reply lte/gateway/c/oai/tasks/amf/amf_app_ue_context_and_proc.h]: Changed the API names accordingly.

**Pull request merged**